examples-gallery/plot_crane_moving_a_load.py

```diff
@@ -0,0 +1,294 @@
+# %%
+"""
+Crane moving a load
+===================
+
+A load is moved by a crane. The load is rotably connected to the crane by a
+massless rod. The goal is to move the load from the initial position to the
+final position in the shortest possible time. The load must not 'overswing'
+the final position. The load is moved by a force applied to its
+suspension point. The jib of the crane is extended in the horizontal
+X direction.
+
+**Constants**
+
+- l: length of the rod attaching the load to the crane [m]
+- m1: mass of mover attached to the arm of the crane [kg]
+- m2: mass of the load [kg]
+- g: acceleration due to gravity [m/s²]
+
+**States**
+
+- xc: x-coordinate of the mover [m]
+- xl: x-coordinate of the load [m]  dependent coordinate
+- yl: y-coordinate of the load [m]  dependent coordinate
+- q: angle of the rod [rad]
+- uxc: velocity of the mover in x-direction [m/s]
+- uxl: velocity of the load in x-direction [m/s] dependent speed
+- uyl: velocity of the load in y-direction [m/s] dependent speed
+- u: angular velocity of the rod [rad/s]
```

Comment:
The states are technically only the minimal set of coordinates and speeds, which are likely q and u.

Reply:
> The states are technically only the minimal set of coordinates and speeds, which are likely q and u.

I did not know this definition, I will correct this and the other suggestions. I needed the dependent coordinates to model the non over . swing.

Reply:
I see below you are doing a non-minimal coordinate formulation. The states are still then only the independent coordinates and speeds.

Reply:
> The states are technically only the minimal set of coordinates and speeds, which are likely q and u.

they are xc, uxc, q, u. xc